livekit-plugins/livekit-plugins-fal/livekit/plugins/fal/stt.py

```diff
@@ -0,0 +1,132 @@
+import dataclasses
+import io
+import os
+import wave
+from dataclasses import dataclass
+
+import fal_client
+from livekit.agents import stt
+from livekit.agents.stt import SpeechEventType, STTCapabilities
+from livekit.agents.utils import AudioBuffer, merge_frames
+
+from .log import logger
+
+
+@dataclass
+class _STTOptions:
+    language: str
+    task: str
+    chunk_level: str
+    version: str
+
+
+class WizperSTT(stt.STT):
+    def __init__(
+        self,
+        *,
+        language="en",
+        task="transcribe",
+        chunk_level="segment",
+        version="3",
+        api_key=None,
+    ):
+        super().__init__(
+            capabilities=STTCapabilities(streaming=False, interim_results=True)
+        )
+        self._api_key = api_key or os.getenv("FAL_API_KEY")
+        self._opts = _STTOptions(
+            language=language, task=task, chunk_level=chunk_level, version=version
+        )
+
+        if not self._api_key:
+            raise ValueError(
+                "FAL AI API key is required. It should be set with env FAL_API_KEY"
+            )
+
+    def _sanitize_options(
+        self,
+        *,
+        language: str = None,
+        task: str = None,
+        chunk_level: str = None,
+        version: str = None,
+    ) -> _STTOptions:
+        config = dataclasses.replace(self._opts)
+        config.language = language or config.language
+        config.task = task or config.task
+        config.chunk_level = chunk_level or config.chunk_level
+        config.version = version or config.version
+        return config
+
+    def _convert_audio_to_wav(self, buffer: AudioBuffer) -> bytes:
+        """Convert AudioBuffer to WAV format bytes."""
+        io_buffer = io.BytesIO()
+        merged_buffer = merge_frames(buffer)
+
+        with wave.open(io_buffer, "wb") as wav:
+            wav.setnchannels(merged_buffer.num_channels)
+            wav.setsampwidth(2)  # 16-bit
+            wav.setframerate(merged_buffer.sample_rate)
+            wav.writeframes(bytes(merged_buffer.data))
+
+        return io_buffer.getvalue()
+
+    # def _create_data_uri(self, wav_bytes: bytes) -> str:
```

Comment:
Curious - did you get a sense for the perf tradeoffs between data URI and uploading/submitting like you do below? My assumption is the latter would be worse for realtime use cases but I'd love your thoughts

Reply:
+1 Agree with that. The less request we make the better :)